Tasks/google-play-release/GooglePlay.js

```diff
@@ -227,29 +239,400 @@ function updateTrack(packageName, track, versionCode, userFraction) {
 /**
  * Add a changelog to an edit
  * Assumes authorized
- * @param {string} changeLogFile - path to changelog file. We assume this exists (behaviour may change)
- * @returns {Promise} track - A promise that will return result from updating a track
+ * @param {string} languageCode Language code (a BCP-47 language tag) of the localized listing to update
+ * @param {string} changelogFile path to changelog file. We assume this exists (behaviour may change)
+ * @returns {Promise} track A promise that will return result from updating a track
  *                            { track: string, versionCodes: [integer], userFraction: double }
  */
-function addChangelog(changeLogFile) {
-    tl.debug("Adding changelog file: " + changeLogFile);
+function addChangelog(languageCode, changelogFile) {
+    tl.debug("Adding changelog file: " + changelogFile);
+
+    var versionCode = globalParams.params.apkVersionCode;
+    try {
+        var changelogVersion = changelogFile.replace(/\.[^/.]+$/g, "");
+        versionCode = parseInt(changelogVersion);
+    } catch (e) {
+        tl.debug(e);
+        tl.debug(`Failed to extract version code from file ${changelogFile}. Defaulting to global version code ${globalParams.params.apkVersionCode}`);
+    }
+
     var requestParameters = {
-        apkVersionCode: globalParams.params.apkVersionCode,
-        language: "en-US",
+        apkVersionCode: versionCode,
+        language: languageCode,
         resource: {
-            language: "en-US",
-            recentChanges: fs.readFileSync(changeLogFile)
+            language: languageCode,
+            recentChanges: fs.readFileSync(changelogFile).toString()
```

Comment:
Wrapped the whole thing in try catch and changed method to reject if we fail to read the file.